### PR TITLE
kustomize: 2.0.3 -> 3.0.0

### PR DESCRIPTION
**Pull request by vdemeester**
###### Motivation for this change

Bump to latest release : [release notes](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/v3.0.0.md).
This also migrate the package to use `buildGoModule` as it is now using go modules.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by vdemeester** [comment]
@GrahamcOfBorg build kustomize

**Review by periklis — APPROVED** [review]
LGTM

**Comment by kalbasit** [comment]
@GrahamcOfBorg build kustomize

File: pkgs/development/tools/kustomize/default.nix [hunk 1]
```diff
@@ -16,11 +17,12 @@ buildGoPackage rec {
   '';
 
   src = fetchFromGitHub {
-    sha256 = "1dfkpx9rllj1bzm5f52bx404kdds3zx1h38yqri9ha3p3pcb1bbb";
+  sha256 = "1ywppn97gfgrwlq1nrj4kdvrdanq5ahqaa636ynyp9yiv9ibziq6";
```
Comment (kalbasit) [inline pkgs/development/tools/kustomize/default.nix]:
```suggestion
    sha256 = "1ywppn97gfgrwlq1nrj4kdvrdanq5ahqaa636ynyp9yiv9ibziq6";
```

File: pkgs/development/tools/kustomize/default.nix [hunk 2]
```diff
@@ -16,11 +17,12 @@ buildGoPackage rec {
...
     rev = "v${version}";
     repo = "kustomize";
     owner = "kubernetes-sigs";
   };
+  modSha256 = "0w8sp73pmj2wqrg7x7z8diglyfq6c6gn9mmck0k1gk90nv7s8rf1";
```
Comment (kalbasit) [inline pkgs/development/tools/kustomize/default.nix]:
```suggestion

  modSha256 = "0w8sp73pmj2wqrg7x7z8diglyfq6c6gn9mmck0k1gk90nv7s8rf1";
```

**Review by kalbasit — CHANGES_REQUESTED** [review]

**Comment by kalbasit** [comment]
@GrahamcOfBorg build kustomize

**Review by kalbasit — APPROVED** [review]
LGTM. It's usually up to the author to rebase the commits. I'll squash merge this time.

**Pull request merged**